activerecord/lib/active_record/associations/preloader.rb

```diff
@@ -84,75 +85,61 @@ class Preloader #:nodoc:
       #   [ :books, :author ]
       #   { author: :avatar }
       #   [ :books, { author: :avatar } ]
-      def preload(records, associations, preload_scope = nil)
-        records = Array.wrap(records).compact
-
-        if records.empty?
-          []
+      def initialize(associate_by_default: true, polymorphic_parent: false, **kwargs)
```

Comment:
As I understood `records` and `associations` are required when using a preloader and it seems 90% of use cases will supply those 2 arguments. Isn't it a better idea to make a signature like `initialize(records = nil, associations =nil, scope: nil, associate_by_default: true, polymorphic_parent: false` ?

Reply:
I prefer this method signature.

Reply:
Don't you think that it will be good idea to raise an error when `records` or `associations` options are not set in this case?